Update LSP4 Metadata for UPs

fhildeb opened the pull request:
### General Improvements
- [x] Reorg Imports
- [x] Simplify code
- [x] Add documentation
- [x] Updated filenames to be more specific

### Added Content
- [x] How to update metadata of a contract using UP

### Tested Scripts
- [x] Tested updated scripts with Maxime

Hugoo commented on smart-contracts-hardhat/scripts/attachAssetMetadataAsUP.ts:
```diff
@@ -0,0 +1,76 @@
+    encodedLSP4Metadata.values[0],
+  ]);
+
+  // Update the ERC725Y storage of the LSP4 metadata
+  const tx = await universalProfile.execute(
```
Maybe:

```suggestion
  // Update the ERC725Y storage of the LSP4 metadata
  // https://docs.lukso.tech/contracts/contracts/ERC725/#execute
  const tx = await universalProfile.execute(
```

Hugoo commented on smart-contracts-hardhat/scripts/getAssetMetadata.ts:
```diff
@@ -1,20 +1,54 @@
+}
+
+// Sample LSP8 NFT
+const SAMPLE_ASSET_ADDRESS = '0x8734600968c7e7193BB9B1b005677B4edBaDcD18';
+
+// Token ID 1 (as Bytes 32)
+const SAMPLE_TOKEN_ID = '0x0000000000000000000000000000000000000000000000000000000000000001';
```
Over time, we get lost with what is what.
I would suggest you to add the network name in the variable or smtgh

Kind of like here:

https://github.com/lukso-network/tools-erc725-inspect/blob/main/constants.ts#L69

or here

https://github.com/lukso-network/tools-erc725-inspect/blob/main/globals.ts#L18

Hugoo commented on smart-contracts-hardhat/scripts/getAssetMetadata.ts:
```diff
@@ -1,20 +1,54 @@
 async function getAssetMetadata() {
-  const token = await ethers.getContractAt(LSP8ABI.abi, ASSET_ADDRESS);
+  // Setup network parameters for ERC725
+  const customNetworkConfig = network.config as CustomNetworkConfig;
+  const networkUrl = customNetworkConfig.url;
+
+  if (!networkUrl) {
+    throw new Error('Network URL is not defined in the Hardhat configuration.');
+  }
+
+  // Create contract instance from ERC725
+  const myAsset = new ERC725(LSP4DigitalAssetSchema, SAMPLE_ASSET_ADDRESS, networkUrl);
```
to follow on previous comment, by reading this it is not clear on which network we are. So the assets address, we don;t know.

The network is selected when we run the hardhat script so this could break somehow if we are on the worng network.

I think it needs more precision

fhildeb commented:
Thanks for the feedback @Hugoo! I've:

- Attached the links to our docs on all function calls (contract or erc725.js)
- Added an enum with a mainnet and testnet address that adjusts based on the selected HardHat network

I also added a `try/catch` block in case there is no sample asset for the current network. 

✅ Tested
